File: MAG/UI/mag_user/lib/models/anime_watchlist.dart
import 'package:json_annotation/json_annotation.dart';

import 'anime.dart';
part 'anime_watchlist.g.dart';

@JsonSerializable()
class AnimeWatchlist {
  int? id;
  int? animeId;
  int? watchlistId;
  String? watchStatus;
  int? progress;
  DateTime? dateStarted;
  DateTime? dateFinished;
  Anime? anime;

  AnimeWatchlist(
    this.id,
    this.animeId,
    this.watchlistId,
    this.watchStatus,
    this.progress,
    this.dateStarted,
    this.dateFinished,
    this.anime,
  );

  factory AnimeWatchlist.fromJson(Map<String, dynamic> json) =>
      _$AnimeWatchlistFromJson(json);

  Map<String, dynamic> toJson() => _$AnimeWatchlistToJson(this);
}

File: MAG/MAG/Database/AnimeWatchlist.cs
﻿using System;
using System.Collections.Generic;

namespace MAG.Database;

public partial class AnimeWatchlist
{
    public int Id { get; set; }

    public int AnimeId { get; set; }

    public int WatchlistId { get; set; }

    public string WatchStatus { get; set; } = null!;

    public int Progress { get; set; }

    public DateTime? DateStarted { get; set; }

    public DateTime? DateFinished { get; set; }

    public virtual Anime Anime { get; set; } = null!;

    public virtual Watchlist Watchlist { get; set; } = null!;
}
